api/src/services/__tests__/competenciesService.ts

```diff
@@ -39,16 +39,24 @@ describe('competenciesService', () => {
   });
 
   describe('createCompetency', () => {
-    it('should create a given competency in the system', async () => {
+    it('should create a competency and model competency in a transaction', async () => {
       const principalId = 2;
       const label = 'label';
       const description = 'description';
       const competencyId = 3;
+      const model_competency_id = 4;
+      mockQuery('BEGIN;');
       mockQuery(
         'insert into `competencies` (`description`, `label`, `principal_id`) values (?, ?, ?)',
```

Comment:
I suppose this doesn't break the code as long as the query values are referenced in the same order, but I'm wondering if there should be some consistencies of how the column names are listed out? In some areas it's `principalId`, then `label` and `description`, and then in some areas, it differs. What do you think?

Reply:
@eileenxue Thanks for pointing that out!  I originally wrote the query in the order that you mentioned.  However, when I ran the tests, I kept on getting "Unexpected Query" error messages.  I then figured out that the columns should be listed alphabetically.  Other `insert` queries in the codebase follow that order too

Reply:
Oh, that's good to know. Thanks for that insightful response, I learned something new!